tests/rules_test.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2022 Jigsaw Operations LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Tests for the RuleSet and Rules classes.
+"""
+
+import ipaddress
+import pytest
+
+from rids import rules
+from rids import observations
+
+
+def test_add_rules():
+  ruleset = rules.RuleSet()
+  r1 = rules.IpRule(
+    msg='test msg',
+    name='test name',
+    url='test url',
+    fetched='fetched when?',
+    matches_ip=ipaddress.ip_address('1.2.3.4'))
+  ruleset.add_ip_rule(r1)
+  ruleset.add_ip_rule(rules.IpRule(
+    msg='test msg',
+    name='test name',
+    url='test url',
+    fetched='fetched later.',
+    matches_ip=ipaddress.ip_address('5.6.7.8')))
+
+
+def _make_ip_rule(ip_address):
+  return rules.IpRule(
+    msg='test msg',
+    name='test name',
+    url='test url',
+    fetched='test fetched time',
+    matches_ip=ipaddress.ip_address(ip_address))
+
+
+def test_match_ip():
+  ruleset = ruleset.RuleSet()
```

Comment:
Change this test to use an `IpMatcher` instead.

Reply:
done